wagtail/test/testapp/urls.py

```diff
@@ -1,7 +1,11 @@
 from django.urls import path
 
+from wagtail.api.v2.router import WagtailAPIRouter
 from wagtail.test.testapp import views
 
+api_router = WagtailAPIRouter("testapp_api_v2")
+api_router.register_endpoint("test_issue_10411", views.Test10411ApiViewSet)
+
```

Comment:
I think we already have a `WagtailAPIRouter` registered in https://github.com/wagtail/wagtail/blob/main/wagtail/test/urls.py for tests, and the API tests should probably be in https://github.com/wagtail/wagtail/tree/main/wagtail/api/v2/tests

Reply:
I struggled a bit with deciding where to register the `Test10411ApiViewSet` endpoint and wouldn't mind making a follow-on PR if this needs fixing.

> I think we already have a WagtailAPIRouter registered in https://github.com/wagtail/wagtail/blob/main/wagtail/test/urls.py for tests

Right; should `Test10411ApiViewSet` simply be registered on `wagtail.test.urls.api_router`?

> and the API tests should probably be in https://github.com/wagtail/wagtail/tree/main/wagtail/api/v2/tests

The test `test_issue_10411` is already in `wagtail/api/v2/tests/test_pages.py`, as you suggest, no? https://github.com/wagtail/wagtail/blob/1ea8a0e3603c8cccbaba03f75b706955d531c999/wagtail/api/v2/tests/test_pages.py#L1029-L1032

Reply:
Thanks @laymonage - I wasn't aware of that. 

@mgax would you be willing to do a new PR to move this test?

Reply:
@lb- Yes, no problem; I just need a bit of clarification, see the comment above.

Reply:
> I struggled a bit with deciding where to register the `Test10411ApiViewSet` endpoint and wouldn't mind making a follow-on PR if this needs fixing.

Yeah the test files may seem like they're a bit scattered around at the moment, but generally I think for optional apps like the API we tend to put the tests and their specific setup in their respective app directory instead of the `testapp`. A follow-up PR would be great, thank you!


> Right; should `Test10411ApiViewSet` simply be registered on `wagtail.test.urls.api_router`?

I think so, if that's possible.

> The test `test_issue_10411` is already in `wagtail/api/v2/tests/test_pages.py`, as you suggest, no?

Yep, but the `Test10411ApiViewSet` should probably be in `wagtail/api/v2/tests/test_pages.py` as well. (Might want to rename it to `Test10411APIViewSet` while you're at it, just for consistency).

Reply:
I've refactored the test in https://github.com/wagtail/wagtail/pull/10647. One check is failing, I wonder if it might be flaky 🤔

Reply:
I've left a https://github.com/wagtail/wagtail/pull/10647#pullrequestreview-1516734824. Don't worry about the failing test, it's failing against Django main after django/django#16920 which means we'll have to update our code.